extensions/ql-vscode/src/remote-queries/sarif-processing.ts

```diff
@@ -186,18 +147,149 @@ export function tryGetRule(
-  const errors = [];
+interface Location {
+  message?: string;
+  filePath: string;
+  codeSnippet: CodeSnippet,
+  highlightedRegion: HighlightedRegion
+}
+
+function validateContextRegion(contextRegion: sarif.Region | undefined): string[] {
+  const errors: string[] = [];
+
+  if (!contextRegion) {
+    errors.push('No context region found in the SARIF result location');
+    return errors;
+  }
+  if (contextRegion.startLine === undefined) {
+    errors.push('No start line set for a result context region');
+  }
+  if (contextRegion.endLine === undefined) {
+    errors.push('No end line set for a result context region');
+  }
+  if (!contextRegion.snippet?.text) {
+    errors.push('No text set for a result context region');
+  }
 
-  if (alert.codeSnippet.startLine > alert.codeSnippet.endLine) {
-    errors.push('The code snippet start line is greater than the end line');
+  if (errors.length > 0) {
+    return errors;
   }
 
-  const highlightedRegion = alert.highlightedRegion;
-  if (highlightedRegion.endLine === highlightedRegion.startLine &&
-    highlightedRegion.endColumn < highlightedRegion.startColumn) {
-    errors.push('The highlighted region end column is greater than the start column');
+  if (contextRegion.startLine! > contextRegion.endLine!) {
+    errors.push('Start line is greater than the end line in result context region');
   }
 
   return errors;
 }
+
+function validateRegion(region: sarif.Region | undefined): string[] {
+  const errors: string[] = [];
+
+  if (!region) {
+    errors.push('No region found in the SARIF result location');
+    return errors;
+  }
+  if (region.startLine === undefined) {
+    errors.push('No start line set for a result region');
+  }
+  if (region.startColumn === undefined) {
+    errors.push('No start column set for a result region');
+  }
+  if (region.endColumn === undefined) {
+    errors.push('No end column set for a result region');
+  }
+
+  if (errors.length > 0) {
+    return errors;
+  }
+
+  if (region.endLine! === region.startLine! &&
+    region.endColumn! < region.startColumn!) {
+    errors.push('End column is greater than the start column in a result region');
+  }
+
+  return errors;
+}
+
+function extractLocation(
+  location: sarif.Location
+): {
+  processedLocation: Location | undefined,
+  errors: string[]
+} {
+  const message = location.message?.text;
+
+  const errors = [];
+
+  const contextRegion = location.physicalLocation?.contextRegion;
+  const contextRegionErrors = validateContextRegion(contextRegion);
+  errors.push(...contextRegionErrors);
+
+  const region = location.physicalLocation?.region;
+  const regionErrors = validateRegion(region);
+  errors.push(...regionErrors);
+
+  const filePath = location.physicalLocation?.artifactLocation?.uri;
+  if (!filePath) {
+    errors.push('No file path found in the SARIF result location');
+  }
+
+  if (errors.length > 0) {
+    return { processedLocation: undefined, errors };
```

Comment:
In this case, I think it would still be nice to return a processed location, but with some fixups. It would be unfortunate if query results couldn't be displayed because the query muddled up some source locations. I think the important thing is to avoid failing when trying to display bad locations and make some sort of best effort to fix up the locations (even if that means reducing the entire source location to a single character. It would still be useful to warn that the source locs are bad, but just do our best to show something.

Reply:
Thank you, I really appreciate these comments - I'm gradually understanding the domain more and more! I'll leave as is for now and we can loop back round with all the other validation bits.